controllers/clusterlogging/clusterlogging_controller.go

```diff
@@ -132,95 +129,31 @@ func (r *ReconcileClusterLogging) updateStatus(instance *loggingv1.ClusterLoggin
-			Name:      constants.SingletonName,
-			Namespace: constants.OpenshiftNS,
-		},
-	}
-}
-
 // SetupWithManager sets up the controller with the Manager.
 func (r *ReconcileClusterLogging) SetupWithManager(mgr ctrl.Manager) error {
-	onAllExceptGenericEventsPredicate := predicate.Funcs{
-		UpdateFunc: func(evt event.UpdateEvent) bool {
-			return true
-		},
-		CreateFunc: func(evt event.CreateEvent) bool {
-			return true
-		},
-		DeleteFunc: func(evt event.DeleteEvent) bool {
-			return true
-		},
-		GenericFunc: func(evt event.GenericEvent) bool {
-			return false
-		},
-	}
-
-	var toElasticsearchRelatedObjRequestMapper handler.MapFunc = func(obj client.Object) []ctrl.Request {
-		isElasticsearchRelatedObj, reconcileRequest := IsElasticsearchRelatedObj(obj)
-		if isElasticsearchRelatedObj {
-			return []ctrl.Request{reconcileRequest}
-		}
-		return []ctrl.Request{}
-	}
-
 	controllerBuilder := ctrl.NewControllerManagedBy(mgr).
-		Watches(&source.Kind{Type: &loggingv1.ClusterLogging{}}, &handler.EnqueueRequestForObject{}).
-		Watches(&source.Kind{Type: &loggingv1.ClusterLogForwarder{}}, &handler.EnqueueRequestForObject{}).
-		Watches(&source.Kind{Type: &corev1.Service{}}, &handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &loggingv1.ClusterLogging{},
-		}).
-		Watches(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &loggingv1.ClusterLogging{},
-		}).
-		Watches(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &loggingv1.ClusterLogging{},
-		}).
-		Watches(&source.Kind{Type: &rbacv1.Role{}}, &handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &loggingv1.ClusterLogging{},
-		}).
-		Watches(&source.Kind{Type: &rbacv1.RoleBinding{}}, &handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &loggingv1.ClusterLogging{},
-		}).
-		Watches(&source.Kind{Type: &corev1.ServiceAccount{}}, &handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &loggingv1.ClusterLogging{},
-		}).
-		Watches(&source.Kind{Type: &appsv1.Deployment{}}, &handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &loggingv1.ClusterLogging{},
-		}).
-		Watches(&source.Kind{Type: &appsv1.DaemonSet{}}, &handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &loggingv1.ClusterLogging{},
-		}).
-		Watches(&source.Kind{Type: &corev1.Secret{}},
-			handler.EnqueueRequestsFromMapFunc(toElasticsearchRelatedObjRequestMapper),
-			builder.WithPredicates(onAllExceptGenericEventsPredicate),
-		)
-
-	return controllerBuilder.
 		For(&loggingv1.ClusterLogging{}).
-		Complete(r)
+		Owns(&corev1.Service{}).
+		Owns(&corev1.ConfigMap{}).
+		Owns(&rbacv1.Role{}).
+		Owns(&rbacv1.RoleBinding{}).
+		Owns(&corev1.ServiceAccount{}).
+		Owns(&appsv1.Deployment{}).
+		Owns(&appsv1.DaemonSet{}).
+		Watches(&source.Kind{Type: &corev1.Secret{}},
```

Comment:
I think we still need an `Owns(&rbacv1.Secret{})` so we receive events for the Secrets that CL owns.